### PR TITLE
Add Windows CI

### PR DESCRIPTION
**Pull request by masterleinad**
We decided that we want to make this work and support MSVC going forward.

### CODE REVIEW

**Comment by aprokop** [comment]
Can you use `[ci skip]` or would that block GitHub actions too?

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      run: |
+        mkdir build
+        cd build
+        cmake \
+          -D CMAKE_BUILD_TYPE=Debug \
+          -D CMAKE_PREFIX_PATH="${KOKKOS_INSTALL_DIR}" \
+          -D ARBORX_ENABLE_MPI=OFF \
+          -D ARBORX_ENABLE_TESTS=ON \
+          -D ARBORX_ENABLE_EXAMPLES=ON \
+          -D ARBORX_ENABLE_BENCHMARKS=OFF \
```
Comment (aprokop) [inline .github/workflows/windows.yml:49]: Why are the benchmarks off? Because `benchmark` not installed?

**Review by aprokop — COMMENTED** [review]
This looks good. Though, how does parallel build and testing work here? I don't see any specifying of the number of parallel tasks.

File: .github/workflows/windows.yml (hunk 1 above)
Comment (masterleinad) [reply .github/workflows/windows.yml:49]: Yes, I haven't bothered with benchmarks yet. Getting this to work should be enough initially and we can always enable more options.

**Comment by masterleinad** [comment]
> Can you use `[ci skip]` or would that block GitHub actions too?

That also affects `GitHub` actions.

**Comment by masterleinad** [comment]
As expected, we get compiler errors such as
```
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2059: syntax error: '<end Parse>' [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2143: syntax error: missing ';' before '{' [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2143: syntax error: missing '>' before ';' [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
  example_dbscan.cpp
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2059: syntax error: '<end Parse>' [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2988: unrecognizable template declaration/definition [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2143: syntax error: missing ';' before '{' [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2059: syntax error: ';' [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2143: syntax error: missing '>' before ';' [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2447: '{': missing function header (old-style formal list?) [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2988: unrecognizable template declaration/definition [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,59): error C2988: unrecognizable template declaration/definition [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2059: syntax error: ';' [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,59): error C2059: syntax error: '||' [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,56): error C2447: '{': missing function header (old-style formal list?) [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(297,31): error C2065: 'Tag': undeclared identifier [D:\a\ArborX\ArborX\build\examples\callback\ArborX_Example_Callback.exe.vcxproj]
D:\a\ArborX\ArborX\src\details\ArborX_DetailsCrsGraphWrapperImpl.hpp(296,59): error C2988: unrecognizable template declaration/definition [D:\a\ArborX\ArborX\build\examples\access_traits\ArborX_Example_HostAccessTraits.exe.vcxproj]
[...]
```
corresponding to https://github.com/arborx/ArborX/blob/c74a6dec6c4acdec22db91573d8cfb99cc6fff6c/src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp#L294-L319.

**Comment by Rombur** [comment]
> As expected, we get compiler errors such as

Yes, @AnhBe0 has a fix for those. We will probably push all the fixes for Windows at once (if there are not too many of them).

**Pull request closed without merging**